Twitter/Twitter_Generate_leads_from_posts.ipynb

```diff
@@ -0,0 +1,329 @@
+    "def get_post_stats(tweetIdList):\n",
+    "    for tweetId in tweetIdList:\n",
+    "\n",
+    "        print(\"*********************************\")\n",
+    "        print(\"Tweet ID : \", tweetId)\n",
+    "        parent_tweet = client.get_tweet(id=tweetId, tweet_fields=[\"author_id\", \"in_reply_to_user_id\",\n",
+    "                                                                             \"referenced_tweets,source,public_metrics\"])\n",
+    "        #parent_tweet_id = reply_tweet.data[\"data\"][\"referenced_tweets\"][0][\"id\"]\n",
+    "        parent_tweet_text = parent_tweet.data[\"text\"]\n",
+    "        parent_public_metrics = parent_tweet.data[\"data\"][\"public_metrics\"]\n",
+    "        parent_retweet_count = parent_public_metrics[\"retweet_count\"]\n",
+    "        parent_reply_count = parent_public_metrics[\"reply_count\"]\n",
+    "        parent_like_count = parent_public_metrics[\"like_count\"]\n",
+    "        parent_quote_count = parent_public_metrics[\"quote_count\"]\n",
+    "        parent_comments_mentions = []\n",
+    "\n",
+    "        print(\"Tweet Text : \", parent_tweet_text)\n",
+    "        print(\"Retweet Count : \", parent_retweet_count)\n",
+    "        retweeted_by_users = client.get_retweeters(id=tweetId)\n",
+    "        \n",
+    "        retweeted_by_username = []\n",
+    "        for retweet_user in retweeted_by_users.data:\n",
+    "            retweeted_by_username.append(retweet_user[\"username\"])\n",
+    "        print(\"Retweeted By : \",retweeted_by_username)\n",
+    "        print(\"Like Count : \", parent_like_count)\n",
+    "        like_by_users = client.get_liking_users(id=tweetId)\n",
+    "        liked_by_username = []\n",
+    "        for like_user in like_by_users.data:\n",
+    "            liked_by_username.append(like_user[\"username\"])\n",
+    "        print(\"Liked By : \", liked_by_username)\n",
+    "        print(\"Reply Count : \", parent_reply_count)\n",
+    "        mentions = client.get_users_mentions(id=user_id)\n",
+    "        replied_by_users = []\n",
+    "        comments = []\n",
+    "        for mention in mentions.data:\n",
+    "            child_tweet = client.get_tweet(id=mention[\"id\"], tweet_fields=[\"author_id\", \"in_reply_to_user_id\",\n",
+    "                                                                                 \"referenced_tweets,source,public_metrics\"])\n",
+    "            in_reply_to_tweet_id = child_tweet.data[\"data\"][\"referenced_tweets\"][0][\"id\"]\n",
+    "\n",
+    "            if str(in_reply_to_tweet_id) == tweetId:\n",
+    "                username = \"@\"+user_name\n",
+    "                temp = str(mention[\"text\"])\n",
+    "                temp = temp.replace(username,\"\")\n",
+    "                comments.append(temp)\n",
+    "        print(\"Comments : \", comments)\n",
+    "\n",
+    "        print(\"Quote Count : \", parent_quote_count)\n",
+    "        print(\" \")\n"
```

Comment:
Should the function `get_post_stats` return a dataframe or something? Seems like it's only printing for now?